onnxruntime/core/providers/cpu/reduction/reduction_ops.cc

```diff
@@ -726,4 +789,10 @@ Status ArgMin<T>::Compute(OpKernelContext* ctx) const {
   return Status::OK();
 }
 
+// Explicit template instantiation
+template class ReduceSum<float>;
+template class ReduceSum<int32_t>;
+template class ReduceSum<double>;
+template class ReduceSum<int64_t>;
+
```

Comment:
Is this needed given we register the ReduceSum op with this class and these types don't we? If needed, should have a comment why only ReduceSum requires this.

Reply:
Without these, I get linker errors. It only passes the linking stage if they are explicitly instantiated. I added a comment.